### PR TITLE
Introduce ingress-perf workload

### PR DESCRIPTION
**Pull request by krishvoor**
This PR aims at integrating the ingress-perf workload to ci-operator

### CODE REVIEW

**Comment by krishvoor** [comment]
Hi @paigerube14 @jtaleric @svetsa-rh 👋 
Request for review

**Comment by krishvoor** [comment]
Hi 👋 @mukrishn ptal

File: ci-operator/step-registry/openshift-qe/ingress-perf/openshift-qe-ingress-perf-ref.yaml
Comment (paigerube14) [inline ci-operator/step-registry/openshift-qe/ingress-perf/openshift-qe-ingress-perf-ref.yaml:13]: Since this commands script will only be running ingress perf tests I think we can hard code/export this in the commands script. 
It would be good to add maybe the config here so if we need to we can run different variations of the test

**Review by paigerube14 — COMMENTED** [review]
You'll want to add a new list item in the _tests_ under the config/openshift-qe folder to actually be able to run the test in the prow system. Currently we have 4.12,4.13,4.14. Would be nice to add in each but we can choose one to start

File: ci-operator/step-registry/openshift-qe/ingress-perf/openshift-qe-ingress-perf-commands.sh
Comment (paigerube14) [reply ci-operator/step-registry/openshift-qe/ingress-perf/openshift-qe-ingress-perf-commands.sh:29]: Also should you take out cluster density v2 as workload since you set workload type already?

Comment (krishvoor) [reply ci-operator/step-registry/openshift-qe/ingress-perf/openshift-qe-ingress-perf-commands.sh:29]: @mukrishn I see these instructions in e2e benchmarking:
https://github.com/cloud-bulldozer/e2e-benchmarking/tree/master/workloads/ingress-perf#ingress-perf

Comment (paigerube14) [reply ci-operator/step-registry/openshift-qe/ingress-perf/openshift-qe-ingress-perf-commands.sh:29]: Looking at the run script I don't think it does anything with that variable

Comment (krishvoor) [reply ci-operator/step-registry/openshift-qe/ingress-perf/openshift-qe-ingress-perf-commands.sh:29]: > Is this config you're using only work for aws?
Not really, though the filename is leading elsewhere, it should work as required across other clouds too.

Comment (krishvoor) [reply ci-operator/step-registry/openshift-qe/ingress-perf/openshift-qe-ingress-perf-commands.sh]: This is now resolved, thank you!

File: ci-operator/step-registry/openshift-qe/ingress-perf/openshift-qe-ingress-perf-ref.yaml
Comment (paigerube14) [inline ci-operator/step-registry/openshift-qe/ingress-perf/openshift-qe-ingress-perf-ref.yaml:12]: Need to remove this line

Comment (svetsa-rh) [reply ci-operator/step-registry/openshift-qe/ingress-perf/openshift-qe-ingress-perf-ref.yaml:12]: @krishvoor If you want, you can add more details to the documentation line that is already at the bottom of this file. 

According to ci-operator syntax, it is expecting documentation at same level as "credentials" and "resources" in the yaml file and does not have context of "documentation" under "credentials". If you check now, you'll notice that 2 of your "mandatory" tests are failing for this reason.

File: ci-operator/step-registry/openshift-qe/ingress-perf/openshift-qe-ingress-perf-ref.yaml
Comment (krishvoor) [reply ci-operator/step-registry/openshift-qe/ingress-perf/openshift-qe-ingress-perf-ref.yaml]: Updated.

File: ci-operator/step-registry/openshift-qe/ingress-perf/openshift-qe-ingress-perf-commands.sh
Comment (paigerube14) [inline ci-operator/step-registry/openshift-qe/ingress-perf/openshift-qe-ingress-perf-commands.sh:26]: can we add the config parameter as an env variable with aws-standard.yml as the default in the ref file

File: ci-operator/step-registry/openshift-qe/ingress-perf/openshift-qe-ingress-perf-commands.sh
Comment (qiliRedHat) [inline ci-operator/step-registry/openshift-qe/ingress-perf/openshift-qe-ingress-perf-commands.sh:23]: The default ES_INDEX for ingress-perf is `ingress-performance`
https://github.com/cloud-bulldozer/e2e-benchmarking/blob/4788805e748cf330953be14a63347f2d3eceb6c2/workloads/ingress-perf/README.md?plain=1#L28

File: ci-operator/step-registry/openshift-qe/ingress-perf/openshift-qe-ingress-perf-commands.sh
Comment (qiliRedHat) [reply ci-operator/step-registry/openshift-qe/ingress-perf/openshift-qe-ingress-perf-commands.sh:26]: Shall we also make ITERATIONS as env variable?

File: ci-operator/step-registry/openshift-qe/ingress-perf/openshift-qe-ingress-perf-commands.sh
Comment (krishvoor) [reply ci-operator/step-registry/openshift-qe/ingress-perf/openshift-qe-ingress-perf-commands.sh:23]: Is the index `ingress-performance` available?

``` 
ES_SERVER: "https://$ES_USERNAME:$ES_PASSWORD@search-ocp-qe-perf-scale-test-elk-hcm7wtsqpxy7xogbu72bor4uve.us-east-1.es.amazonaws.com" 
```

Comment (krishvoor) [reply ci-operator/step-registry/openshift-qe/ingress-perf/openshift-qe-ingress-perf-commands.sh:23]: Ack - will stick with `ingress-performance` as the index.

Comment (krishvoor) [reply ci-operator/step-registry/openshift-qe/ingress-perf/openshift-qe-ingress-perf-commands.sh]: Updated.

**Comment by paigerube14** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.12-nightly-ocp-qe-perfscale-ci-tests-rosa-ingress-perf

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.12-nightly.yaml
Comment (svetsa-rh) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.12-nightly.yaml:55]: @krishvoor Please update the test name here to "perfscale-rosa-multiaz-ingress-perf-3nodes". I will update remaining test names soon. I have been working mostly on 4.13 yaml file and haven't gotten to others yet, but we will stick to the test names that are noted in 4.13 rosa test suite yml fille.

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.12-nightly.yaml
Comment (svetsa-rh) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.12-nightly.yaml:56]: Are we going to stick with 3 nodes testing for ingress perf or do we plan to scale? 
Not sure if this is relevant thread, but could you ptal https://redhat-internal.slack.com/archives/GTHU29259/p1688584111328949

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.12-stable-prod.yaml
Comment (svetsa-rh) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.12-stable-prod.yaml:36]: @krishvoor Any reason why we want to run these tests in prod? Sorry for any confusion, this test suite was added to specifically add any tests that need to run in prod as exception, otherwise, we always run in stage only as of now. Alternatively, we could leave the test here and remove cron, but set "always_run: false" and keep the test for adhoc testing. See https://github.com/openshift/release/blob/master/ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.13-nightly.yaml#L24 for example. 
I will update all 4.12 tests to avoid confusion.
FYI, this change will require you to rerun "make jobs".

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.13-nightly.yaml
Comment (svetsa-rh) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.13-nightly.yaml:249]: Same question here, REPLICAS=3? Can you ptal at Sachin's and Joe's comments here https://redhat-internal.slack.com/archives/GTHU29259/p1688584111328949
I am not sure what the conclusion (if any) there is.

**Review by svetsa-rh — CHANGES_REQUESTED** [review]
@krishvoor I added few comments. Can you ptal?

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.12-stable-prod.yaml
Comment (krishvoor) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.12-stable-prod.yaml:36]: >we always run in stage only as of now.
Ack, updating the params, included ingress-job to reflect across prod/ candidate/ nightly releases

Comment (krishvoor) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.12-stable-prod.yaml]: Added.

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.12-nightly.yaml
Comment (krishvoor) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.12-nightly.yaml:56]: For now, we can stick with 3 node testing.

We have baselined with different instance sizes and node-labeling.
which I'd like to tackle in a different PR & figure out ci-operator docs on how to achieve

Comment (svetsa-rh) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.12-nightly.yaml:56]: Harsha, if you see other Rosa examples, you will see various env variables on how to specify instance sizes, zones etc.. I haven't tried node labeling yet.

Comment (krishvoor) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.12-nightly.yaml:56]: Yes, re: Instance Sizes I observed in other ROSA examples.
Resolving this conversation for now.

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.12-nightly.yaml
Comment (krishvoor) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.12-nightly.yaml:55]: I missed this, pushing changes.

Comment (krishvoor) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.12-nightly.yaml]: Updated, ptal

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.13-nightly.yaml
Comment (qiliRedHat) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.13-nightly.yaml:249]: @svetsa-rh This pr is for ingress-perf, Sachin and Joe's comments should be for k8s-netperf.

Comment (qiliRedHat) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.13-nightly.yaml:249]: @svetsa-rh I'm ok with this PR, I'll leave it to you to add lgtm.

**Comment by paigerube14** [comment]
This looks good to me! I'll wait for @svetsa-rh review

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.12-nightly.yaml
Comment (svetsa-rh) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.12-nightly.yaml:55]: Since this is a cron, can you please add "periodic" at the end of the test name? You can see other examples. Other than that PR lgtm

**Review by svetsa-rh — CHANGES_REQUESTED** [review]

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.12-nightly.yaml
Comment (krishvoor) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.12-nightly.yaml]: Sure, updated, ptal

**Comment by svetsa-rh** [comment]
@paigerube14 @jtaleric This needs approve label.

File: ci-operator/jobs/openshift/release/openshift-release-master-presubmits.yaml
Comment (paigerube14) [inline ci-operator/jobs/openshift/release/openshift-release-master-presubmits.yaml:435]: @krishvoor can you take this out? I'm not sure how it got here but its causing approval needed from outside our repo owners

Comment (krishvoor) [reply ci-operator/jobs/openshift/release/openshift-release-master-presubmits.yaml]: Hey @paigerube14 can you ptal?
I had to rebase and re-run `make update` locally, which updated
`[ci-operator/jobs/openshift/release/openshift-release-master-presubmits.yaml]`

**Review by paigerube14 — APPROVED** [review]
/lgtm

**Comment by jmguzik** [comment]
This PR potentially blocks the batch to get tested properly

**Pull request merged**
